### PR TITLE
Fixed Email Backend

### PR DESCRIPTION
**Pull request by brettallenyo**
Fixed email backend to be able to send emails from an appspot email address.

Also altered the backend so that instead of taking raw HTML it takes information from the user like latitude, longitude, itemTypes, and all of the stores to construct server-side HTML that is placed inside of the email.

### CODE REVIEW

File: capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java [hunk 1]
```diff
@@ -40,61 +43,105 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+    message.setFrom(new InternetAddress("shopsmart@step39-2020.appspotmail.com", "Shopsmart"));
     // Set To: header field of the header.
-    message.addRecipient(Message.RecipientType.TO, new InternetAddress(to));
+    message.addRecipient(Message.RecipientType.TO, new InternetAddress(email));
     // Set Subject: header field
     message.setSubject("Message From ShopSmart");
     // Now set the actual message
-    message.setContent(html, "text/html");
+    message.setContent(storesHTML, "text/html");
     System.out.println("sending...");
```
Comment (chen-dawn) [inline capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java:78]: Remove print statement.

File: capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java [hunk 2]
```diff
@@ -40,61 +43,105 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
     Transport.send(message);
   }
 
+  private String getStoresHTML(
+      List<Store> stores, List<String> itemTypes, double latitude, double longitude) {
+    String url = "https://step39-2020.uc.r.appspot.com/stores/?";
+    String items = itemTypes.get(0);
+    boolean first = true;
+    for (String item : itemTypes) {
+      url = url + "items=" + item + "&";
```
Comment (chen-dawn) [inline capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java:89]: Preferably use a url builder, like: https://github.com/googleinterns/step39-2020/blob/b6bd5778b874021e9c25ac89b9b105d57ef65404/capstone/backend/src/main/java/com/google/servlets/GetStoreRankingsServlet.java#L179

File: capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java [hunk 3]
```diff
@@ -40,61 +43,105 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+            + "latitude="
+            + latitude
+            + "&longitude="
+            + longitude
+            + "&distanceValue=2147483647&method=location";
+    String storesString = "";
+    for (Store store : stores) {
+      storesString =
+          storesString
+              + "<h3>"
```
Comment (chen-dawn) [inline capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java:103]: This is okay, though what's probably cleaner is if you have something like 
```
String htmlTemplate = "<h3>$storeName</h3><p>Total Items Found: $itemsFound </p>";
String formattedString= htmlTemplate
                                          .replace("$storeName", store.getStoreName())
                                          .replace("$itemsFound", store.getNumberOfItemsFound());
```

File: capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java [hunk 4]
```diff
@@ -40,61 +43,105 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
     System.out.println("sending...");
     // Send message
     Transport.send(message);
   }
 
+  private String getStoresHTML(
+      List<Store> stores, List<String> itemTypes, double latitude, double longitude) {
+    String url = "https://step39-2020.uc.r.appspot.com/stores/?";
+    String items = itemTypes.get(0);
+    boolean first = true;
```
Comment (luli0521) [inline capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java:87]: Where is it used? Remove if unused.

File: capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java [hunk 5]
```diff
@@ -40,61 +43,105 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+    message.setContent(storesHTML, "text/html");
     System.out.println("sending...");
     // Send message
     Transport.send(message);
   }
 
+  private String getStoresHTML(
+      List<Store> stores, List<String> itemTypes, double latitude, double longitude) {
+    String url = "https://step39-2020.uc.r.appspot.com/stores/?";
+    String items = itemTypes.get(0);
```
Comment (luli0521) [inline capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java:86]: You put the item at index 0 twice. Here you get it and in the for loop below, you get it for another time.

File: capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java [hunk 6]
```diff
@@ -40,61 +44,105 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+      List<Store> stores, List<String> itemTypes, double latitude, double longitude) {
+    String url = "https://step39-2020.uc.r.appspot.com/";
+    try {
+      URIBuilder urlb = new URIBuilder("https://step39-2020.uc.r.appspot.com/stores/");
+      for (String item : itemTypes) {
+        urlb.addParameter("items", item);
+      }
+      urlb.addParameter("latitude", "" + latitude);
+      urlb.addParameter("longitude", "" + longitude);
+      urlb.addParameter("distanceValue", "2147483647");
```
Comment (chen-dawn) [inline capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java:93]: what is this distance value?

Comment (brettallenyo) [reply capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java:93]: It is the value for preferred distance when we allowed users to set their preference before, but now we just automatically set it to the max apparently

File: capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java (hunk 3 above)
Comment (brettallenyo) [reply capstone/backend/src/main/java/com/google/servlets/ShareViaEmailServlet.java:104]: Will do this on next fix

**Pull request merged**